lib/client.js

```diff
@@ -0,0 +1,169 @@
+var WS = require('ws');
+var debug = require('debug')('docker-exec-websocket-server:lib:client');
+var url = require('url');
+var events = require('events');
+var _ = require('lodash');
+var assert = require('assert');
+var msgcode = require('../lib/messagecodes.js');
+var through = require('through');
+
+export default class DockerExecWebsocketClient extends events.EventEmitter {
+  constructor (options) {
+    super();
+    this.options = _.defaults({}, options, {
+      tty: true,
+      command: ['/bin/bash'],
+      protocol: 'ws',
+      slashes: true,
+    });
+  }
```

Comment:
When default arguments are set can't we do things like:
`assert(this.options.command instanceof Array, "options.command must be an array");`
Just to protect people against stupid bugs...
Also check that `url` is given and that it starts with either `ws://` or `wss://`, I think :)